tests/test_validation.py

```diff
@@ -0,0 +1,2803 @@
+
+    return dict
+
+class TestAllClear0(unittest.TestCase):
+    
+
+    def load_verbosity(self):
+        self.verbosity = utility_get_verbosity()
+    
+    
+    def step_0(self): 
+        validate.prepare_outdirs()
+        self.energy_channel = {'min': 10, 'max': -1, 'units': 'MeV'}
+        self.energy_key = objh.energy_channel_to_key(self.energy_channel)
+        self.all_energy_channels = [self.energy_key] 
+        self.model_names = ['Test_model_0']
+        observation_json = './tests/files/observations/validation/all_clear/all_clear_false.json'
+        observation = utility_load_observation(observation_json, self.energy_channel)
+        observation_objects = {self.energy_key: [observation]}
+        self.verbosity = utility_get_verbosity()
+        forecast_json = './tests/files/forecasts/validation/all_clear/pred_all_clear_false.json'
+        forecast = utility_load_forecast(forecast_json, self.energy_channel)
+        forecast_objects = {self.energy_key: [forecast]}
+        self.sphinx, self.obs_thresholds, self.obs_sep_events = utility_match_sphinx(self.all_energy_channels,\
+             self.model_names, observation_objects, forecast_objects)
+        self.profname_dict = None
+        self.DoResume = False
+        
+        self.validation_quantity = ['all_clear', 'awt', 'duration', 'end_time', 'fluence', 'max_flux_in_pred_window', 'peak_intensity_max', 'peak_intensity_max_time', 'peak_intensity' \
+            'peak_intensity_time', 'probability', 'start_time', 'threshold_crossing', 'time_profile']
+        self.quantities_tested = ['all_clear', 'awt']
+        self.validation_type = ['All']
+        
+    def step_1(self):
+       
+        self.dataframe = validate.fill_sphinx_df(self.sphinx, self.model_names, self.all_energy_channels, \
+            self.obs_thresholds, self.profname_dict)
+        for keywords in self.dataframe:
+            # temp = self.sphinx['Test_model_0'][self.energy_key].prediction.short_name\
+            
+            temp = attributes_of_sphinx_obj(keywords, self.sphinx['Test_model_0'][self.all_energy_channels[0]][0],\
+                 self.energy_key, self.obs_thresholds)
+            if pd.isnull(temp) and pd.isnull(self.dataframe[keywords][0]):
+                self.assertTrue(pd.isnull(self.dataframe[keywords][0]))
+            else:
+                self.assertEqual(self.dataframe[keywords][0], temp, 'Error is in keyword ' + keywords)
+
+    def step_2(self):
+        validate.calculate_intuitive_metrics(self.dataframe, self.model_names, self.all_energy_channels, \
+                self.obs_thresholds, 'All')
+    
+
+    def step_3(self):
+        test_dict = initialize_all_clear_dict()
+        test_dict = fill_all_clear_dict_hit(test_dict, self)
+        
+        csv_filename = './tests/output\\csv\\all_clear_metrics.csv'
```

Comment:
Mixed Mac/Linux file path convention / with Windows convention \.  Use Mac/Linux convention, exclusively.  Better yet, use `os.path.join()` to avoid using path separators at all: `os.path.join(config.outdir, 'csv', 'all_clear_metrics.csv')`